pkg/controller/nstemplateset/nstemplateset_controller.go

```diff
@@ -0,0 +1,353 @@
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusNamespaceProvisionFailed, err,
+			"failed to provision namespace '%s'", nsName)
+	}
+
+	objs = template.Filter(objs, template.RetainNamespaces)
+	for _, rawObj := range objs {
+		obj := rawObj.Object
+		if nsObj, ok := obj.(*unstructured.Unstructured); ok {
+			// set labels
+			labels := nsObj.GetLabels()
+			if labels == nil {
+				labels = make(map[string]string)
+			}
+			labels["owner"] = username
+			nsObj.SetLabels(labels)
+
+			// set owner ref
+			if err := controllerutil.SetControllerReference(nsTmplSet, nsObj, r.scheme); err != nil {
+				return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusNamespaceProvisionFailed, err,
+					"failed to set controller reference for namespace '%s'", nsName)
+			}
+		}
+	}
+
+	err = tmplProcessor.Apply(objs)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusNamespaceProvisionFailed, err,
+			"failed to provision namespace '%s'", nsName)
+	}
+
+	log.Info("namespace provisioned", "namespace", tcNamespace)
+	if err := r.setStatusProvisioning(nsTmplSet); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileNSTemplateSet) ensureOtherResources(logger logr.Logger, nsTmplSet *toolchainv1alpha1.NSTemplateSet, tcNamespace *toolchainv1alpha1.Namespace, params map[string]string) error {
+	username := nsTmplSet.GetName()
+	nsName := ToNamespaceName(username, tcNamespace.Type)
+
+	tmplContent, err := templates.GetTemplateContent(fmt.Sprintf("%s-%s", nsTmplSet.Spec.TierName, tcNamespace.Type))
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusNamespaceProvisionFailed, err,
+			"failed to provision namespace '%s'", nsName)
+	}
+
+	tmplProcessor := template.NewProcessor(r.client, r.scheme)
+	objs, err := tmplProcessor.Process(tmplContent, params)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusNamespaceProvisionFailed, err,
+			"failed to provision namespace '%s'", nsName)
+	}
+	objs = template.Filter(objs, template.RetainAllButNamespaces)
+	err = tmplProcessor.Apply(objs)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusNamespaceProvisionFailed, err,
+			"failed to provision namespace '%s'", nsName)
+	}
+
+	namespace := &corev1.Namespace{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: nsName}, namespace); err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusNamespaceProvisionFailed, err, "failed to get namespace '%s'", nsName)
+	}
+	if namespace.Labels == nil {
+		namespace.Labels = make(map[string]string)
+	}
+	namespace.Labels["revision"] = tcNamespace.Revision
+	if err := r.client.Update(context.TODO(), namespace); err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, nsTmplSet, r.setStatusNamespaceProvisionFailed, err, "failed to update namespace '%s'", nsName)
+	}
```

Comment:
Looks like we are verifying namespace under verification of other resources. this is already done in `ensureNamespaceResources`. Do we really need this? https://github.com/codeready-toolchain/member-operator/pull/53/files#diff-cffea82f7fdc7427b8c9f0d0fdd55aedR230-R239

Reply:
Not getting you, this is about update_namespace with `revison` label to mark that the provisioning is done for this namespace.  It's nothing related to verifying_namespace.